include/zephyr/arch/arm/cortex_m/cpu.h

```diff
@@ -52,18 +52,25 @@ extern "C" {
 #ifdef CONFIG_PM_S2RAM
 
 struct __cpu_context {
-	/* GPRs are saved onto the stack */
+	/* GPRs are saved on stack */
 	uint32_t msp;
-	uint32_t msplim;
 	uint32_t psp;
-	uint32_t psplim;
 	uint32_t apsr;
 	uint32_t ipsr;
 	uint32_t epsr;
 	uint32_t primask;
-	uint32_t faultmask;
-	uint32_t basepri;
 	uint32_t control;
+
+#	ifdef CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+	/* Registers present on ARMv7-M and ARMv8-M */
+	uint32_t basepri;
+	uint32_t faultmask;
+#		ifdef CONFIG_ARMV8_M_MAINLINE
+	/* Registers present only on ARMv8-M */
+	uint32_t msplim;
+	uint32_t psplim;
+#		endif
+#	endif
```

Comment:
The ifdef/endif alignment is incorrect.

Reply:
How so? Note that the `#endif` for `#ifdef CONFIG_PM_S2RAM` is later down the file.

Reply:
Sorry, i meant the white-spacing. According to the code-styling rules it is my understanding that every preprocessing-directive start at the beginning of the line and should not have whitespaces between the hashtag and the tag.

Reply:
I did a search, and to my surprise could not find anything about this (I did not even know this was possible). Both zephyr and linux style-guide do not mention it and the clangd parser also does not complain. I can even see in some libraries the same usage:

```
#if OPENSSL_VERSION_NUMBER >= 0x10100000L /* 1.1.0 */
      SSL_CTX_set_min_proto_version(backend->ctx, SSL3_VERSION);
      SSL_CTX_set_max_proto_version(backend->ctx, SSL3_VERSION);
#else
      ctx_options |= SSL_OP_NO_SSLv2;
      ctx_options |= SSL_OP_NO_TLSv1;
#  if OPENSSL_VERSION_NUMBER >= 0x1000100FL
      ctx_options |= SSL_OP_NO_TLSv1_1;
      ctx_options |= SSL_OP_NO_TLSv1_2;
#    ifdef TLS1_3_VERSION
      ctx_options |= SSL_OP_NO_TLSv1_3;
#    endif
#  endif
#endif
```

But: i could not find this setup in the zephyr kernel itself, and the usage in the libraries / external sources shows much shorter indents: 1 or 2 spaces per indent instead of 4. With only 1 or 2 spaces indent, the directive is still visually (easy) detectable as directive. With the 4-spaces the directive is much harder to spot between the lines of code.

So, as I'm not part of the zephyr-team, i cannot say a lot about it, but my guess is that - as I could not find this approach in the kernel - the zephyr-team does not use it. And if it is used, the indent is only 1-2 spaces to keep the directive readable / easy recognisable.